kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/checks/utils/Scheduler.kt

```diff
@@ -0,0 +1,90 @@
+package com.kotlindiscord.kord.extensions.checks.utils
+
+import kotlinx.coroutines.*
+import mu.KotlinLogging
+import kotlin.coroutines.EmptyCoroutineContext
+
+private val logger = KotlinLogging.logger {  }
+
+/**
+ * Class in charge of providing scheduling functions.
+ */
+class Scheduler {
+    private val jobMap: MutableMap<Int, Job> = mutableMapOf()
+
+    private val scope = CoroutineScope(EmptyCoroutineContext)
+    private val finishTask = CancellationException()
+
+    /**
+     * Schedule a new task.
+     *
+     * @param id The ID of the task.
+     * @param delay How long in milliseconds to wait before executing the callback.
+     * @param data Arbitrary data to be passed to the callback.
+     * @param callback Function to be executed.
+     *
+     * @return Return true on success, false otherwise.
+     */
+    fun <T> schedule(id: Int, delay: Long, data: T?, callback: (T?) -> Unit): Boolean {
```

Comment:
I don't think we really want users to have to specify their own ID. You could use a UUID instead, and return it - since we'd only ever return `false` here if the ID is already scheduled.

If you can think of a legitimately useful reason to be able to provide a specific ID for a job here, then you can create an overloaded method that does that.

Also, how about scheduling repeating jobs?

Reply:
Then this should return this ID, otherwise, it's unable to cancel/complete jobs.

Reply:
Yup, that's what I said!